src/datachain/query/session.py

```diff
@@ -154,16 +155,23 @@ def get_or_create_job(self) -> "Job":
                 script = str(uuid4())
             python_version = f"{sys.version_info.major}.{sys.version_info.minor}"
 
-            # try to find the parent job
-            parent = self.catalog.metastore.get_last_job_by_name(script)
+            # Determine parent job based on DATACHAIN_CHECKPOINTS_RESET flag
+            checkpoints_reset = env2bool("DATACHAIN_CHECKPOINTS_RESET", undefined=False)
+            if checkpoints_reset:
+                # User wants fresh start - don't link to parent
+                parent_job_id = None
+            else:
+                # Normal run - try to find parent job for checkpoint reuse
+                parent = self.catalog.metastore.get_last_job_by_name(script)
```

Comment:
do we have a checkbox to implement a thing with caching job id (or whatever idea we had) to avoid collisions across CLI/CICD on Studio  runs?